### ISSUE TITLE
[Request] Public request handlers

### ISSUE THREAD

**Issue by kevelbreh** [square/picasso#885]
tl;dr, Is there any reason why the following handlers are internal?

* [AssetRequestHandler](https://github.com/square/picasso/blob/master/picasso/src/main/java/com/squareup/picasso/AssetRequestHandler.java)
* [ContactsPhotoRequestHandler](https://github.com/square/picasso/blob/master/picasso/src/main/java/com/squareup/picasso/ContactsPhotoRequestHandler.java)
* [ContentStreamRequestHandler](https://github.com/square/picasso/blob/master/picasso/src/main/java/com/squareup/picasso/ContentStreamRequestHandler.java)
* [MediaStoreRequestHandler](https://github.com/square/picasso/blob/master/picasso/src/main/java/com/squareup/picasso/MediaStoreRequestHandler.java)
* [NetworkRequestHandler](https://github.com/square/picasso/blob/master/picasso/src/main/java/com/squareup/picasso/NetworkRequestHandler.java)
* [ResourceRequestHandler](https://github.com/square/picasso/blob/master/picasso/src/main/java/com/squareup/picasso/ResourceRequestHandler.java)
 
Surely making these handlers publicly available would allow for better extension when it comes to requiring similar implementation of the handler, with one or two alterations.  The set of handlers provides a solid core of functionality already. Having to extend the request handler itself just to re-implement what has already been done, and then some custom logic; is a bit of an overkill.

**Comment by JakeWharton** [comment]
They are implantation details that we reserve the right to change at any
time. Moreover, you can simply copy their contents if you want to create
your own with similar functionality. If they were in the public API we
would have to ensure API stability and compatibility which we are not
prepared to do.

**Comment by JakeWharton** [comment]
If you can make a strong and concrete argument (preferably with examples) for this we would consider it. But thus far everyone just gives hypothetical reasons which would serve only to massively increase the burden of maintaining the library with little (to no) actual utility.

**Comment by posativ** [comment]
> Moreover, you can simply copy their contents if you want to create
your own with similar functionality.

Which requires: Stats, Downloader, a few non-public fields in Response, Utils.createDefaultDownloader. Roughly just to chain e.g. custom request handler and network handler.
